docs/standards/smart-contracts/lsp2-json-schema-utils.md

```diff
@@ -0,0 +1,339 @@
+---
+title: LSP2JSONSchemaUtils
+sidebar_position: 12
+---
+
+# LSP2JSONSchemaUtils
+
+:::info
+
+[`LSP2Utils.sol`](https://github.com/lukso-network/lsp-smart-contracts/blob/develop/contracts/LSP2ERC725YJSONSchema/LSP2Utils.sol)
+
+:::
+
+This library should be used to generate ERC725Y keys described in the JSON Schema.
+
+## Functions
+
+### generateSingletonKey
+
+```solidity
+function generateSingletonKey(
+    string memory keyName
+) internal pure returns (bytes32);
+```
+
+Generates a data key of `keyType` Singleton.
+
+#### Parameters:
+
+| Name      | Type   | Description                                          |
+| :-------- | :----- | :--------------------------------------------------- |
+| `keyName` | string | The string to hash to generate a Singleton data key. |
+
+#### Return Values:
+
+| Name     | Type    | Description                      |
+| :------- | :------ | :------------------------------- |
+| `result` | bytes32 | Data key of `keyType` Singleton. |
+
+### generateArrayKey
+
+```solidity
+function generateArrayKey(
+    string memory keyName
+) internal pure returns (bytes32);
+```
+
+Generates a data key of `keyType` Array by hashing `keyName`.
+
+#### Parameters:
+
+| Name      | Type   | Description                                                            |
+| :-------- | :----- | :--------------------------------------------------------------------- |
+| `keyName` | string | The string that will be used to generate a data key of `keyType` Array |
+
+#### Return Values:
+
+| Name     | Type    | Description                  |
+| :------- | :------ | :--------------------------- |
+| `result` | bytes32 | Data key of `keyType` Array. |
+
+:::note
+
+#### Requirements:
+
+- The last two characters should be `[]`.
+  E.g.:
+
+```solidity
+string memory keyName = "ArrayName[]";
+```
+
+:::
+
+### generateArrayElementKeyAtIndex
+
+```solidity
+function generateArrayElementKeyAtIndex(
+    bytes32 arrayKey,
+    uint256 index
+) internal pure returns (bytes32);
+```
+
+Generates an Array index data key by concatenating the first 16 bytes of `arrayKey` and `index` transformed from uint256 to bytes16 (uint256 -> uint128 -> bytes16)
```

Comment:
I don't think we should explain the underlying code (in the function body), but just roughly what is happening. I suggest this.

```suggestion
Generates an Array index data key by concatenating the first 16 bytes of `arrayKey` with an `index`. 
```